presto-main/src/main/java/io/prestosql/sql/analyzer/Field.java

```diff
@@ -23,74 +23,76 @@
-    private final Optional<String> originColumnName;
     private final Optional<QualifiedName> relationAlias;
     private final Optional<String> name;
     private final Type type;
     private final boolean hidden;
     private final boolean aliased;
+    private final Optional<Source> origin;
 
     public static Field newUnqualified(String name, Type type)
     {
         requireNonNull(name, "name is null");
         requireNonNull(type, "type is null");
 
-        return new Field(Optional.empty(), Optional.of(name), type, false, Optional.empty(), Optional.empty(), false);
+        return new Field(Optional.empty(), Optional.of(name), type, false, Optional.empty(), false);
     }
 
     public static Field newUnqualified(Optional<String> name, Type type)
     {
         requireNonNull(name, "name is null");
         requireNonNull(type, "type is null");
 
-        return new Field(Optional.empty(), name, type, false, Optional.empty(), Optional.empty(), false);
+        return new Field(Optional.empty(), name, type, false, Optional.empty(), false);
     }
 
-    public static Field newUnqualified(Optional<String> name, Type type, Optional<QualifiedObjectName> originTable, Optional<String> originColumn, boolean aliased)
+    public static Field newUnqualified(Optional<String> name, Type type, Optional<Source> origin, boolean aliased)
     {
         requireNonNull(name, "name is null");
         requireNonNull(type, "type is null");
-        requireNonNull(originTable, "originTable is null");
+        requireNonNull(origin, "origin is null");
 
-        return new Field(Optional.empty(), name, type, false, originTable, originColumn, aliased);
+        return new Field(Optional.empty(), name, type, false, origin, aliased);
     }
 
-    public static Field newQualified(QualifiedName relationAlias, Optional<String> name, Type type, boolean hidden, Optional<QualifiedObjectName> originTable, Optional<String> originColumn, boolean aliased)
+    public static Field newQualified(QualifiedName relationAlias, Optional<String> name, Type type, boolean hidden, Optional<Source> origin, boolean aliased)
     {
         requireNonNull(relationAlias, "relationAlias is null");
         requireNonNull(name, "name is null");
         requireNonNull(type, "type is null");
-        requireNonNull(originTable, "originTable is null");
+        requireNonNull(origin, "originTable is null");
 
-        return new Field(Optional.of(relationAlias), name, type, hidden, originTable, originColumn, aliased);
+        return new Field(Optional.of(relationAlias), name, type, hidden, origin, aliased);
     }
 
-    public Field(Optional<QualifiedName> relationAlias, Optional<String> name, Type type, boolean hidden, Optional<QualifiedObjectName> originTable, Optional<String> originColumnName, boolean aliased)
+    public static Field newSourceField(QualifiedName relationAlias, QualifiedObjectName table, String name, Type type, boolean hidden)
```

Comment:
newQualified? The method above can also create a field with a source (origin relation).

Make order of parameters to be coherent with other factory methods.

The best it would be to extract the builder.